exonum/src/runtime/dispatcher/mod.rs

```diff
@@ -44,13 +44,68 @@ mod tests;
 struct ServiceInfo {
     runtime_id: u32,
     name: String,
+    status: InstanceStatus,
+}
+
+/// Lookup table for the committed service instances.
+#[derive(Debug, Default)]
+struct CommittedServices {
+    instances: BTreeMap<InstanceId, ServiceInfo>,
+    instance_names: BTreeMap<String, InstanceId>,
```

Comment:
Are keys of `instance_names` need to be ordered? Otherwise, I'd use `HashMap` (AFAIK, it's faster).